### PR TITLE
Minor syntax changes

### PR DESCRIPTION
**Pull request by dhsorens**
## Overview
1. We port any reference of syntax to the syntax module because K5 was complaining. We hope to put it back eventually.
2. We make one change in the comment notation in `specific-matching-rules.k` for readability.
3. Finally, we change the syntax for unforgeable names from `#( !I:Int )` to `unforgeable( !I:Int )` for readability.

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.
Add link to corresponding JIRA issue.

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You assigned one person to review this PR

### If you are not a member of the core development team, please confirm:
- [ ] You signed the commit. Merging requires a signature. Please see the [developer wiki](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain) for instructions.
- [ ] Your GitHub account is also an account with [Travis CI](https://travis-ci.org). Unit tests will not run on your PR unless you have an account with Travis. Merge requires passed unit tests.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: rholang/src/main/k/rholang/free.k [hunk 1]
```diff
@@ -4,34 +4,6 @@ imports JOINS
...
-            //                                        ]{ Set of free vars }[ Node constructed (T/F)]
-    | "[" Int "][" Exp "][" Int ";" HigherPats "][" List ";" HigherPats "]{" SetOrList "}[" Bool "]"
-            // We need to make sure that a binder doesn't try to bind two variables twice. To do
-            // this, we put a special marker on those elements corresponding to a binder
-          //| "[" Int "][" HigherPat "][" Exp "][" List "]{" SetOrList "}{" List "}{""}[" Bool "]"
-
-
 
 // ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* ******* //
 /// The "fv( )" function: collects free variables of any statement
```
Comment (differentialderek) [inline rholang/src/main/k/rholang/free.k]: Only changing location of syntax.

File: rholang/src/main/k/rholang/matching-function.k [hunk 2]
```diff
@@ -11,55 +11,6 @@ imports MATCH-PROCESS
...
-// Syntax which is needed for various matching operations
-syntax CanBeStdMatched ::= Id
-              | "intostdmatchform(" HigherPat ")"
-              | "intostdtfmatchform(" HigherPat ")"
-              | String StringlessMatchFormSyntax
-              | Ground
-              | HigherNameNoVars
-              | KResult
-
 // Now the semantics of the match function
```
Comment (differentialderek) [inline rholang/src/main/k/rholang/matching-function.k]: Only changing location of syntax.

File: rholang/src/main/k/rholang/matching-with-par.k [hunk 3]
```diff
@@ -5,18 +5,6 @@ imports SPECIFIC-MATCHING-RULES
...
-syntax PatternMatchingCells ::=
-   "[" String "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" Int "]["
-                                                                           HigherProcPat "]{"Set"}"
- | "[" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "]"
-     // For variable matching at the end
- | "[" HigherProcPat "][" HigherProcPat "][" HigherProcPat "][" HigherProcPat "]["
-                                                                  HigherProcPat "]{"HigherProcPat"}"
-
-
 // If we get a pattern or a process sent that has a "|" in it, we pause computation and invoke
```
Comment (differentialderek) [inline rholang/src/main/k/rholang/matching-with-par.k]: Only changing location of syntax.

File: rholang/src/main/k/rholang/rho-syntax.k [hunk 4]
```diff
@@ -321,4 +321,108 @@ syntax HigherMatchCasePat  ::= HigherProcPat "=>" "{" HigherProcPat "}" [binder]
...
+             "[" Int "][" HigherPat "][" Exp "][" List "]{" SetOrList "}[" Bool "]"
+            // Nodes corresponding to binders (bnodes) are given the syntax
+            // [ Id ][ Parent ][ Plus ID ; Plus Process/Name ][ Minus ID ; Minus Process/Name
+            //                                        ]{ Set of free vars }[ Node constructed (T/F)]
+          | "[" Int "][" Exp "][" Int ";" HigherPats "][" List ";" HigherPats "]{" SetOrList "}[" Bool "]"
+            // We need to make sure that a binder doesn't try to bind two variables twice. To do
+            // this, we put a special marker on those elements corresponding to a binder
+          //| "[" Int "][" HigherPat "][" Exp "][" List "]{" SetOrList "}{" List "}{""}[" Bool "]"
+
 endmodule
```
Comment (differentialderek) [inline rholang/src/main/k/rholang/rho-syntax.k]: Each syntax block that was removed is put together here, under appropriate headings.

File: rholang/src/main/k/rholang/sending-receiving.k [hunk 5]
```diff
@@ -4,9 +4,9 @@ imports PROCESSES-SEMANTICS
 // *******************************************************************
 //                        SENDING-RECEIVING
 // *******************************************************************
-// Some syntax specifically for the "sending-receiving" module
-// for the data we need to keep on our in and out cells
-syntax InOutData ::= Int ";" Set
+
+
+
```
Comment (differentialderek) [inline rholang/src/main/k/rholang/sending-receiving.k]: Only changing location of syntax.

File: rholang/src/main/k/rholang/specific-matching-rules.k [hunk 6]
```diff
@@ -81,9 +85,12 @@ rule <thread> ... <k> ... match["TFSTDMATCH"]( P:HigherPat ; X:Wildcard )
...
+// *******************************************************************
+// MATCHING NAMES
 
+// * * * * * * * * * * * * * * * * * * * *
 // The "DoTheseHigherNamesMatch" function
-// This function will see
+
 // 1. We allow variables in the second component
 rule <thread> ... <k> ... match["HIGHERNAMEMATCH"]( P:HigherPat ; D:HigherPat )
               => matchstdform["HIGHERNAMEMATCH"]( intostdmatchform(P) ; intostdmatchform(D))
```
Comment (differentialderek) [inline rholang/src/main/k/rholang/specific-matching-rules.k]: New headings make it slightly more readable.

**Review by ghost — APPROVED** [review]

**Pull request merged**